appcues/src/main/java/com/appcues/statemachine/SideEffect.kt

```diff
@@ -4,5 +4,5 @@ import com.appcues.action.ActionProcessor
 
 internal interface SideEffect {
 
-    suspend fun launch(processor: ActionProcessor): Action?
+    suspend fun launch(duringTransition: Transition, processor: ActionProcessor): Action?
```

Comment:
I like the idea but I think there should be a way for us to infer the transition (current state + side effect) without needing to pass it in the launch. maybe we can have ReportError(shouldRetry = true) and if this is the next action (in the state machine take logic) then we can move to FailingState(Transition(currentState, sideEffect)) to avoid passing this "duringTransition" to every sideEffect.

the idea of this launch methods to provide a way for "outside dependencies" like action processor to be used by side effects.

Reply:
@andretortolano this is all updated now to remove this extra `duringTransition` param, like we chatted about. Thanks for the ideas!